Firestore/core/test/firebase/firestore/remote/datastore_test.mm

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/remote/datastore.h"
+
+#include "Firestore/core/src/firebase/firestore/util/async_queue.h"
+#include "Firestore/core/src/firebase/firestore/util/executor_std.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+using core::DatabaseInfo;
+using model::DatabaseId;
+using util::AsyncQueue;
+using util::internal::ExecutorStd;
+
+namespace {
+
+class NoOpObserver : public GrpcStreamObserver {
+ public:
+  void OnStreamStart() override {
```

Comment:
totally optional: add debug logging in each of the event handler for helping debug?

Reply:
I'd prefer to keep this observer as simple as possible (I'm only defining this class because the stream expects a non-null observer). I'm concerned about duplication with `grpc_stream_tests.cc`, which does have an observer that keeps track of which callbacks were invoked.